v4/core/base_service_retries_test.go

```diff
@@ -0,0 +1,461 @@
+// +build all slow basesvc retries
+
+package core
+
+// (C) Copyright IBM Corp. 2020.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+// assertResponse is a convenience function for checking various parts of a response
+func assertResponse(r *DetailedResponse, expectedStatusCode int, expectedContentType string) {
+	Expect(r).ToNot(BeNil())
+	Expect(r.StatusCode).To(Equal(expectedStatusCode))
+	if expectedContentType != "" {
+		Expect(r.Headers.Get("Content-Type")).To(Equal(expectedContentType))
+	}
+}
+
+// clientInit is a convenience function for setting up a new service and request builder
+func clientInit(method string, url string, maxRetries int, maxIntervalSecs int) (service *BaseService, builder *RequestBuilder) {
+	var err error
+	options := &ServiceOptions{
+		URL:           url,
+		Authenticator: &NoAuthAuthenticator{},
+	}
+	service, err = NewBaseService(options)
+	Expect(err).To(BeNil())
+	if maxRetries > 0 {
+		service.EnableRetries(maxRetries, time.Duration(maxIntervalSecs)*time.Second)
+	}
+	builder = NewRequestBuilder(method)
+	_, err = builder.ConstructHTTPURL(url, nil, nil)
+	Expect(err).To(BeNil())
+	return
+}
+
+var _ = Describe(`Retry scenarios`, func() {
+	var server *httptest.Server
+
+	BeforeEach(func() {
+		SetLogger(NewLogger(LevelDebug, log.New(GinkgoWriter, "", log.LstdFlags)))
```

Comment:
This step will create a new go core logger at "debug" level and use the test frameworks io.Writer instance as the output "stream".   This means that when you run the tests, you'll see the debug messages if you run with the `-ginkgo.v` option (e.g. `go test -ginkgo.v -tags=retries`).